#! /usr/bin/env python
# Forearm camera configuration

PACKAGE='ardrone_pose_estimation'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("Position_P",double_t, SensorLevels.RECONFIGURE_STOP,    "Position P gain. The default value is 0.23",0.23,0.0,1)
gen.add("Position_I",double_t, SensorLevels.RECONFIGURE_STOP,    "Position I gain. The default value is 0.01",0.01,0.0,1)
gen.add("Position_D",double_t, SensorLevels.RECONFIGURE_STOP,    "Position D gain. The default value is 0.26",0.26,0.0,1)


gen.add("Velocity_P",double_t, SensorLevels.RECONFIGURE_STOP,    "Velocity P gain. The default value is 0.2",0.2,0.0,2)
gen.add("Velocity_I",double_t, SensorLevels.RECONFIGURE_STOP,    "Velocity I gain. The default value is 0.0",0.0,0.0,2)
gen.add("Velocity_D",double_t, SensorLevels.RECONFIGURE_STOP,    "Velocity D gain. The default value is 0.12",0.12,0.0,2)

exit(gen.generate(PACKAGE, "ardrone_pose_estimation", "onlinegain"))
